tests/next_tests/iterator_tests/test_symbol_ref_utils.py

```diff
@@ -0,0 +1,60 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from dataclasses import dataclass
+from typing import Optional
+
+from gt4py import eve
+from gt4py.next.iterator import ir as itir
+from gt4py.next.iterator.transforms.symbol_ref_utils import (
+    collect_symbol_refs,
+    get_user_defined_symbols,
+)
+
+
+def test_get_user_defined_symbols():
+    @dataclass
+    class _GetRealItirSymTable(eve.VisitorWithSymbolTableTrait):
+        extracted_symtable: Optional[dict] = None
+
+        @classmethod
+        def apply(cls, ir: itir.Node):
+            obj = cls()
+            obj.visit(ir)
+            return obj.extracted_symtable
+
+        def visit_SymRef(self, node: itir.SymRef, **kwargs):
+            self.extracted_symtable = kwargs["symtable"]
+
+    ir = itir.FencilDefinition(
+        id="foo",
+        function_definitions=[],
+        params=[itir.Sym(id="target_symbol")],
+        closures=[
+            itir.StencilClosure(
+                domain=itir.FunCall(fun=itir.SymRef(id="cartesian_domain"), args=[]),
+                stencil=itir.SymRef(id="deref"),
+                output=itir.SymRef(id="target_symbol"),
+                inputs=[],
+            )
+        ],
+    )
+    testee = _GetRealItirSymTable.apply(ir)
```

Comment:
```suggestion
    testee = GetSymtableForSymref.apply(ir, itir.SymRef(id="target_symbol"))
```
Just a suggestion, but it would be easier to digest if the symref you want the symtable for is explicitly given. Is the `_GetRealItirSymTable` actually needed or can't you just use `ir.annex["symtable"]`?

Reply:
I want to test that this pass extracts _only_ the user defined symrefs, therefore I want the real symbol table as a pass would give you (this symtable will contain the builtins). The symtable annex should not be used outside of a pass as recently discussed here https://github.com/GridTools/gt4py/pull/1137/files. So I am not sure if you didn't understand my test or if I didn't understand your comment.

Reply:
fixed